[rules_cc] Remove deprecated cc_proto_library shim

Reasoning for not upstreaming this patch: Drake-specific tweak to
avoid a dependency on protobuf.

--- cc/defs.bzl
+++ cc/defs.bzl
@@ -13,7 +13,6 @@
 # limitations under the License.
 """Starlark rules for building C++ projects."""
 
-load("@com_google_protobuf//bazel:cc_proto_library.bzl", _cc_proto_library = "cc_proto_library")
 load("//cc:cc_binary.bzl", _cc_binary = "cc_binary")
 load("//cc:cc_import.bzl", _cc_import = "cc_import")
 load("//cc:cc_library.bzl", _cc_library = "cc_library")
@@ -43,11 +42,7 @@
 
 # DEPRECATED: use rule from com_google_protobuf repository
 def cc_proto_library(**kwargs):
+    pass
-    if "deprecation" not in kwargs:
-        _cc_proto_library(deprecation = "Use cc_proto_library from com_google_protobuf", **kwargs)
-    else:
-        _cc_proto_library(**kwargs)
-
 # Toolchain rules
 
 cc_toolchain = _cc_toolchain
